src/main/java/org/opensearch/neuralsearch/processor/normalization/MaxNormalizationTechnique.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.processor.normalization;
+
+import lombok.ToString;
+import lombok.extern.log4j.Log4j2;
+import org.apache.lucene.search.ScoreDoc;
+import org.apache.lucene.search.TopDocs;
+import org.opensearch.neuralsearch.processor.CompoundTopDocs;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+@ToString(onlyExplicitlyIncluded = true)
+@Log4j2
+public class MaxNormalizationTechnique implements ScoreNormalizationTechnique {
+
+    @ToString.Include
+    public static final String TECHNIQUE_NAME = "max_technique";
+
+    @Override
+    public void normalize(List<CompoundTopDocs> queryTopDocs) {
+        int numOfSubqueries = queryTopDocs.stream()
+            .filter(Objects::nonNull)
+            .filter(topDocs -> topDocs.getTopDocs().size() > 0)
+            .findAny()
+            .get()
+            .getTopDocs()
+            .size();
+        if (numOfSubqueries < 2 || numOfSubqueries > 2) {
+            throw new IllegalArgumentException("Number of subqueries cannot be grater or lessor than 2");
+        }
+
+        float[] maxScoresPerSubquery = getMaxScores(queryTopDocs, numOfSubqueries);
+
+        float matchQueryMaxScore = maxScoresPerSubquery[0];
+        log.info("max score of match query [{}]", matchQueryMaxScore);
+        float knnQueryMaxScore = maxScoresPerSubquery[1];
```

Comment:
I assume this is the requirement for them to send knn query always at the second position, but here we're relying on user's good intentions

Reply:
So for the POC purposes, I have made it a compulsion that KNN query should be a second subquery for this normlization technique. This we can convey to the user for testing purposes to always provide knn query as 2nd subquery. Here the intention is to make this POC ready for running a few benchmarks.

Reply:
This is a fair trade-off for POC. Since testing is done by us to evaluate the technique.